src/resolvers/findMany.ts

```diff
@@ -58,15 +58,15 @@ export function findMany<TSource = any, TContext = any, TDoc extends Document =
     throw new Error('First arg for Resolver findMany() should be instance of Mongoose Model.');
   }
 
-  if (!tc || tc.constructor.name !== 'ObjectTypeComposer') {
+  if (!tc || !(tc instanceof ObjectTypeComposer)) {
     throw new Error('Second arg for Resolver findMany() should be instance of ObjectTypeComposer.');
   }
 
   const aliases = prepareNestedAliases(model.schema);
   const aliasesReverse = prepareAliasesReverse(model.schema);
 
   return tc.schemaComposer.createResolver<TSource, TArgs>({
-    type: tc.NonNull.List.NonNull,
+    type: tc.schemaComposer.getAnyTC(tc.getTypeName()).NonNull.List.NonNull,
```

Comment:
That would definitely be a much cleaner change for now.

I definitely agree on the changing the typing to allow interface TCs to resolvers as a long term solution, especially since it seems that interfaces don't cause issues when 'hacked' in as a string type currently. 
My original intention with these changes was to avoid messing with existing code and types as much as possible to avoid accidentally breaking anything, though until I read through what the old DTCs did I was thinking something like this would be needed anyway. 

Passing in an Interface TC directly to the resolver (and presumably letting it be passed through composeMongoose() and convertModelToGraphQL() as needed) means we should be able to cut down on code inside eDTC for dealing with the interface. Perhaps EDTC should extend InterfaceTypeComposer() instead of OTC in that case, and we could just have an Object TC as a property for creating and manipulating the custom input type. (like how we have now as eDTC.DInputObject) 

Currently for top level model discriminators each sub-model being created using composeMongoose() instead of convertModelToGraphQL() so they have a set of mongoose resolvers generated. (since they have a full mongoose model already stored on the main model which can be used by the resolvers)
```typescript
// foreach discrimTC
const discrimTC =
        discrimType instanceof Schema
          ? convertModelToGraphQL(
              { schema: discrimType },
              this.getTypeName() + key,
              schemaComposer,
              { ...this.opts, name: this.getTypeName() + key }
            )
          : composeMongoose(discrimType as Model<any>, { ...this.opts, name: this.getTypeName() + key });
```

 These can then be accessed and used by calling getDisciminatorTCs() which ensures resolvers are present for better typing when importing and using it
```typescript
  getDiscriminatorTCs(): {
    [key: string]: ObjectTypeComposer<any, TContext> & {
      mongooseResolvers: GenerateResolverType<any, TContext>;
    };
  } {
    // check if mongooseResolvers are present (assume on one = on all)
    if ((this.discrimTCs[Object.keys(this.discrimTCs)[0]] as any).mongooseResolvers) {
      return this.discrimTCs as any;
    } else {
      return {};
    }
  }
```

This is roughly how I've been using it in my test setup:
```typescript
...
const composeOpts: ComposeMongooseOpts = {
  includeBaseDiscriminators: true,
  includeNestedDiscriminators: true,
};

export const PClass = mongoose.model('ClassGraph', pClass);
const ExtraClass = PClass.discriminator('extraToken', classSecondSchema);
export const ClassTC = composeMongoose(PClass, composeOpts);

const ExtraClassTC = (ClassTC as EDiscriminatorTypeComposer<any, any>).getDiscriminatorTCs()
  ?.extraToken;
export { ExtraClassTC };

```
in a 'routes.js' file then:
```typescript
schemaComposer.Query.addFields({
  classById: ClassTC.mongooseResolvers.findById(),
  extraClassById: ExtraClassTC.mongooseResolvers.findById()
}
```

I'd like to get rid of the need to use the (... as EDTC) if discriminators are enabled in the options to use EDTC only props/methods, not sure if there is a way to do that in typescript.

Reply:
> Maybe better to change `'[' + tc.getTypeName() + '!]!'` if you want to override type in future.
> 
> But I'm still unsure that use type as a string is a good idea. Maybe we just allow passing Interfaces to all resolvers:
> 
> ```diff
> export function findMany<TSource = any, TContext = any, TDoc extends Document = any>(
>   model: Model<TDoc>,
> -  tc: ObjectTypeComposer<TDoc, TContext>,
> +  tc: ObjectTypeComposer<TDoc, TContext> | InterfaceTypeComposer<TDoc, TContext>,
>   opts?: FindManyResolverOpts
> ):
> ```
> 
> `Resolvers` in graphql-compose === FieldConfigs` in graphql.
> So we can generate as many Resolvers as we want. And maybe we just need generate a new resolver for every Discriminator type?

@nodkz Is there any progress on allowing InterfaceTCs directly in resolvers, or should I continue with this PR as is?

Reply:
@jhbuchanan45 I've just updated master which allows passing InterfaceTCs to resolvers. Please take a look.

Reply:
@nodkz I've pushed an updated version of the eDTC which now extends the InterfaceTypeComposer and is passed directly. It's definitely a neater solution, and it was a lot less work to allow IFTCs as well as OTCs than I had expected. 

Cheers for making those changes to the resolvers, I had mistakenly assumed that it would need changes in graphql-compose.